src/asm/output.c

```diff
@@ -0,0 +1,83 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   output.c                                           :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: asarandi <asarandi@student.42.us.org>      +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2018/02/11 20:52:17 by asarandi          #+#    #+#             */
+/*   Updated: 2018/02/28 01:31:21 by asarandi         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "asm.h"
+
+int		update_acb(int acb, int index, int op_type)
+{
+	int	tmp;
+	int shift;
+
+	tmp = 0;
+	shift = 0;
+	if (op_type == T_REG)
+		tmp = REG_CODE;
+	if (op_type == T_DIR)
+		tmp = DIR_CODE;
+	if (op_type == T_IND)
+		tmp = IND_CODE;
+	if (index == 0)
+		shift = 6;
+	if (index == 1)
+		shift = 4;
+	if (index == 2)
+		shift = 2;
+	tmp = (tmp << shift);
+	return (acb + tmp);
```

Comment:
Addition is dangerous here - what if update_acb() gets called twice with the same index?
A safer way to write this would be:

 `acb = (acb &~ (3 << shift)) | (tmp << shift);`.